### PR TITLE
feat: create DispatchTable UI component (#21)

### LINKED ISSUE #21: Create DispatchTable UI component
## Description
- Implement `lib/ui/components/DispatchTable.jsx`
- Use ink-table or custom Ink Box layout
- Columns: Project, Issue/PR, Branch, Status, Age
- Status icons per brand palette
- Props: dispatches array, selectedIndex (for highlighting)

## Acceptance Criteria
- [ ] Table renders with all columns
- [ ] Selection highlighting works
- [ ] Status icons display correctly

**Size:** M | **Owner:** Kaylee | **Depends on:** #1

### PR DESCRIPTION
**Pull request by jsturtevant**
Closes #21

## Changes
- **lib/ui/components/DispatchTable.jsx**: Ink component rendering dispatch data as a table with columns (Project, Issue/PR, Branch, Status, Age), status icons per brand palette, selectedIndex-based row highlighting (inverse styling), and formatAge utility
- **lib/ui/index.js**: Barrel export for UI components (DispatchTable, STATUS_ICONS, formatAge)
- **test/jsx-loader.mjs**: ESM loader hook using esbuild to transform .jsx files for Node.js test runner
- **test/ui/DispatchTable.test.js**: 12 tests covering header rendering, data rows, status icons, selection highlighting, empty state, and age formatting
- **package.json**: Added esbuild devDep, updated test script to run UI tests with JSX loader

## Acceptance Criteria
- [x] Table renders with all columns (Project, Issue/PR, Branch, Status, Age)
- [x] Selection highlighting works (selected row uses inverse styling)
- [x] Status icons display correctly (🔵 planning, 🟡 reviewing, 🟢 active, ✅ done, 🔴 blocked, ❌ error)

## Test Results
All 240 tests pass (228 existing + 12 new).

### CODE REVIEW

**Review by jsturtevant — COMMENTED** [review]
**Mal Review — PR #45: DispatchTable UI Component**

LGTM — all acceptance criteria met. Clean implementation.

**Checklist:**
- ✅ All 5 columns render (Project, Issue/PR, Branch, Status, Age)
- ✅ Selection highlighting via inverse styling works
- ✅ All 6 status icons mapped correctly
- ✅ Age formatting covers minutes/hours/days + edge cases (null, future)
- ✅ 12 real tests using ink-testing-library — covers headers, data rows, icons, selection, empty state, formatAge
- ✅ JSX + ES modules, proper React/Ink imports
- ✅ No .squad/ files in diff
- ✅ esbuild JSX loader is a pragmatic approach for Node test runner

Two minor nits inline (non-blocking). **Approved** (would mark APPROVE but GitHub prevents self-approval on own PR).

File: lib/ui/components/DispatchTable.jsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+      {/* Data rows */}
+      {rows.length === 0 ? (
+        <Box>
+          <Text dimColor>No active dispatches</Text>
+        </Box>
+      ) : (
+        rows.map((row, i) => (
+          <TableRow key={i} cells={row} selected={i === selectedIndex} />
+        ))
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply lib/ui/components/DispatchTable.jsx]: Fixed — using session_id as stable key with index fallback.
